conformance/echo-basic/echo-basic.go

```diff
@@ -200,12 +208,95 @@ func runH2CServer(h2cPort string, errchan chan<- error) {
 	}
 }
 
+// Global variable to store the SNI retrieved at a different level.
+var globalsni string
```

Comment:
This variable being global may cause race conditions when the tests are being run parallelly.

Reply:
Ok thanks, changed it.

Reply:
All changes to echo-basic.go have been reverted.